src/platform_impl/windows/event_loop.rs

```diff
@@ -543,115 +599,108 @@ lazy_static! {
     // WPARAM and LPARAM are unused.
     static ref USER_EVENT_MSG_ID: u32 = {
         unsafe {
-            winuser::RegisterWindowMessageA("Winit::WakeupMsg\0".as_ptr() as LPCSTR)
+            RegisterWindowMessageA(PSTR("Winit::WakeupMsg".as_ptr() as *mut u8))
```

Comment:
@kennykerr Could the non-std variant perhaps provide a non-std zero-cost `IntoParam` implementation (I think asked this elsewhere) that:

- Is implemented for `&str` and `&[u8]` (`b"byte literals")`;
- Asserts there's a trailing null (which got lost in the last commit);
- Take care of mutability conversion.

Reply:
Sounds reasonable. Please create an issue if there isn't already one.

Reply:
Perhaps something like this: https://github.com/microsoft/windows-rs/issues/1307

Reply:
@kennykerr A macro may be nice, but a (`const`?) constructor that borrows the (`'static`) literal (by returning `fn new(&[u8]) -> &PSTR`) and asserts on an existing trailing null would be nice too.

Reply:
Yep I can add that - the macro is come compelling for wide string literals.